Can't upgrade to 0.1.646-beta from 0.1.635-beta

## Actual behavior

Looks like this is the actual bug:

```
warning CS8785: Generator 'SourceGenerator' failed to generate source. It will not contribute to the output and compilation errors may occur as a result. Exception was of type 'IOException' with message 'The process cannot access the file 'C:\Users\virzak\.nuget\packages\microsoft.windows.sdk.win32metadata\17.0.2-preview\Windows.Win32.winmd' because it is being used by another process.'
```

## Expected behavior

A clear and concise description of what you expected to happen.

## Repro steps

1. `NativeMethods.txt` content:
```
GetWindowLong
SetWindowLong
WINDOW_STYLE
```

### Context

- CsWin32 version: 0.1.646-beta
- Target Framework: net6
- `LangVersion` preview (C# 11)

--- a/src/Microsoft.Windows.CsWin32/MetadataIndex.cs
+++ b/src/Microsoft.Windows.CsWin32/MetadataIndex.cs
@@ -250,11 +250,12 @@ namespace Microsoft.Windows.CsWin32
                 // Read the entire metadata file exactly once so that many MemoryStreams can share the memory.
                 if (!MetadataFiles.TryGetValue(metadataPath, out MemoryMappedFile? file))
                 {
-                    file = MemoryMappedFile.CreateFromFile(metadataPath);
+                    FileStream metadataStream = new FileStream(metadataPath, FileMode.Open, FileAccess.Read, FileShare.Read);
+                    file = MemoryMappedFile.CreateFromFile(metadataStream, mapName: null, capacity: 0, MemoryMappedFileAccess.Read, HandleInheritability.None, leaveOpen: false);
                     MetadataFiles.Add(metadataPath, file);
                 }
 
-                metadataBytes = file.CreateViewStream();
+                metadataBytes = file.CreateViewStream(offset: 0, size: 0, MemoryMappedFileAccess.Read);
             }
 
             return new MetadataIndex(metadataPath, metadataBytes, platform);

--- a/test/Microsoft.Windows.CsWin32.Tests/GeneratorTests.cs
+++ b/test/Microsoft.Windows.CsWin32.Tests/GeneratorTests.cs
@@ -2429,6 +2429,14 @@ namespace Windows.Win32
         Assert.Contains(classDecl.AttributeLists, al => al.Attributes.Any(a => a.Name.ToString().Contains("ComImport")));
     }
 
+    [Fact]
+    public void OpensMetadataForSharedReading()
+    {
+        using FileStream competingReader = File.OpenRead(MetadataPath);
+        this.generator = this.CreateGenerator();
+        Assert.True(this.generator.TryGenerate("CreateFile", CancellationToken.None));
+    }
+
     private static string ConstructGlobalConfigString(bool omitDocs = false)
     {
         StringBuilder globalConfigBuilder = new();

